overlord/overlord.go

```diff
@@ -41,9 +47,25 @@ type Overlord struct {
 func New() (*Overlord, error) {
 	o := &Overlord{}
 
-	// TODO: read it or create a fresh one and learn about the system
-	// current state
-	s := state.New(nil)
+	backend := state.NewStateFsBackend(dirs.SnapStateFile)
```

Comment:
I suggest moving all of that new logic into a new local private function:

`func loadState() (*state.State, error)`